PLUGINS.md

```diff
@@ -0,0 +1,476 @@
+        pass
+
+    def register_track_metadata_processor(function: Callable, priority: PluginPriority = PluginPriority.NORMAL) -> None:
+        pass
+
+    # Event hooks
+    def register_album_post_removal_processor(function: Callable, priority: PluginPriority = PluginPriority.NORMAL) -> None:
+        pass
+
+    def register_file_post_load_processor(function: Callable, priority: PluginPriority = PluginPriority.NORMAL) -> None:
+        pass
+
+    def register_file_post_addition_to_track_processor(function: Callable, priority: PluginPriority = PluginPriority.NORMAL) -> None:
+        pass
+
+    def register_file_post_removal_from_track_processor(function: Callable, priority: PluginPriority = PluginPriority.NORMAL) -> None:
+        pass
+
+    def register_file_post_save_processor(function: Callable, priority: PluginPriority = PluginPriority.NORMAL) -> None:
+        pass
+
+    # Cover art
+    def register_cover_art_provider(provider: CoverArtProvider) -> None:
+        pass
+
+    # File formats
+    def register_format(format: File) -> None:
+        pass
+
+    # Scripting
+    def register_script_function(function: Callable, name: str = None, eval_args: bool = True,
+                                 check_argcount: bool = True, documentation: str = None) -> None:
+        pass
+
+    # Context menu actions
+    def register_album_action(action: BaseAction) -> None:
+        pass
+
+    def register_cluster_action(action: BaseAction) -> None:
+        pass
+
+    def register_clusterlist_action(action: BaseAction) -> None:
+        pass
+
+    def register_track_action(action: BaseAction) -> None:
+        pass
+
+    def register_file_action(action: BaseAction) -> None:
+        pass
+
+    # UI
+    def register_options_page(page_class: Type[OptionsPage]) -> None:
+        pass
+
+    # TODO: Replace by init function in plugin
+    # def register_ui_init(function: Callable) -> None:
+    #     pass
+
+    # Other ideas
+    # Implement status indicators as an extension point. This allows plugins
+    # that use alternative progress displays
+    # def register_status_indicator(function: Callable) -> None:
+    #     pass
+
+    # Register page for file properties. Same for track and album
+    # def register_file_info_page(page_class):
+    #     pass
+
+    # For the media player toolbar?
+    # def register_toolbar(toolbar_class):
+    #     pass
+```
+
+
+### Localization (l10n) and internationalization (i18n)
+
+TBD
+
+
+### Plugin life cycle
+
+TBD
+
+
+## Distribution
+
+In order to both simplify third-party development of plugins and distribute
+the maintenance work there will no longer be a single plugin repository. Instead
+each plugin SHOULD be provided in a separate git repository. Plugins also CAN
+be installed locally without a git repository by placing the plugin package
+inside the plugin directory.
+
+
+### Repository structure
+
+A Picard plugin repository MUST be a git repository containing exactly one
```

Comment:
Not exactly one, as discussed previously, we'll support one ore more plugins per repo for convenience.